core/store/src/flat/types.rs

```diff
@@ -142,5 +177,20 @@ pub struct FetchingStateStatus {
     pub num_parts: u64,
 }
 
+/// Current step of resharding flat storage - splitting parent.
+#[derive(
+    BorshSerialize, BorshDeserialize, Clone, Debug, PartialEq, Eq, serde::Serialize, ProtocolSchema,
+)]
+pub struct SplittingParentStatus {
+    /// UId of the left child shard. Will contain everything lesser than boundary account.
+    pub left_child_shard: ShardUId,
+    /// UId of the right child shard. Will contain everything greater or equal than boundary account.
+    pub right_child_shard: ShardUId,
+    /// The new shard layout.
+    pub shard_layout: ShardLayout,
```

Comment:
I was on the fence about storing `shard_layout` here, but it makes things so much more self-contained in `FlatStorageResharder` and testing easier as well.